src/SimplePhotometryModel.cc

```diff
@@ -106,5 +80,73 @@ PhotometryMappingBase *SimplePhotometryModel::findMapping(CcdImage const &ccdIma
     return i->second.get();
 }
 
+SimpleFluxModel::SimpleFluxModel(CcdImageList const &ccdImageList) : SimplePhotometryModel(ccdImageList) {
+    for (auto const &ccdImage : ccdImageList) {
+        auto photoCalib = ccdImage->getPhotoCalib();
+        // Use the single-frame processing calibration from the PhotoCalib as the initial value.
+        auto transfo = std::make_shared<FluxTransfoSpatiallyInvariant>(photoCalib->getCalibrationMean());
+        _myMap.emplace(ccdImage->getHashKey(),
+                       std::unique_ptr<PhotometryMapping>(new PhotometryMapping(transfo)));
+    }
+    LOGLS_INFO(_log, "SimpleFluxModel got " << _myMap.size() << " ccdImage mappings.");
+}
+
+double SimpleFluxModel::computeResidual(CcdImage const &ccdImage, MeasuredStar const &measuredStar) const {
+    return transform(ccdImage, measuredStar) - measuredStar.getFittedStar()->getFlux();
+}
+
+double SimpleFluxModel::transform(CcdImage const &ccdImage, MeasuredStar const &star) const {
+    auto mapping = findMapping(ccdImage);
+    return mapping->transform(star, star.getInstFlux());
+}
+
+double SimpleFluxModel::transformError(CcdImage const &ccdImage, MeasuredStar const &star) const {
+    auto mapping = findMapping(ccdImage);
+    return mapping->transformError(star, star.getInstFluxErr());
+}
+
+std::shared_ptr<afw::image::PhotoCalib> SimpleFluxModel::toPhotoCalib(CcdImage const &ccdImage) const {
+    double calibration = (findMapping(ccdImage)->getParameters()[0]);
+    auto oldPhotoCalib = ccdImage.getPhotoCalib();
+    return std::unique_ptr<afw::image::PhotoCalib>(
+            new afw::image::PhotoCalib(calibration, oldPhotoCalib->getCalibrationErr()));
```

Comment:
So jointcal isn't updating the estimated uncertainty in the calibration?

Reply:
Also, we're in C++14 now, so you can use `std::make_unique()`.

Reply:
Correct: I haven't worked through the math about what the new PhotoCalib uncertainty is. The input uncertainty for that CCD Exposure's Calib object is close enough for the moment, I think?

Reply:
Yeah, this is not a priority for photometry until we get the constrained photometry fit working.  And I suppose we don't actually have a place to put the astrometric uncertainty, though maybe we should think about how to propagate that as well.

Reply:
Agreed on all points.